roles/openshift_hosted/tasks/router/router.yml

```diff
@@ -9,10 +9,15 @@
     module: slurp
     src: "{{ item }}"
   register: openshift_router_certificate_output
+  # Defaulting dictionary keys to none to avoid deprecation warnings
+  # (future fatal errors) during template evaluation. Dictionary keys
+  # won't be accessed unless openshift_hosted_router_certificate is
+  # defined and has all keys (certfile, keyfile, cafile) which we
+  # check above.
   with_items:
-  - "{{ openshift_hosted_router_certificate.certfile }}"
-  - "{{ openshift_hosted_router_certificate.keyfile }}"
-  - "{{ openshift_hosted_router_certificate.cafile }}"
+  - "{{ (openshift_hosted_router_certificate | default({'certfile':none})).certfile }}"
+  - "{{ (openshift_hosted_router_certificate | default({'keyfile':none})).keyfile }}"
+  - "{{ (openshift_hosted_router_certificate | default({'cafile':none})).cafile }}"
   when: openshift_hosted_router_certificate is defined
```

Comment:
I remember seeing something nearly identical to what we're doing. I'll follow up with it.